core/src/services/azblob/core.rs

```diff
@@ -370,6 +374,119 @@ impl AzblobCore {
         Ok(req)
     }
 
+    pub fn azblob_put_block_request(
+        &self,
+        path: &str,
+        block_id: Uuid,
+        size: Option<u64>,
+        args: &OpWrite,
+        body: AsyncBody,
+    ) -> Result<Request<AsyncBody>> {
+        // To be written as part of a blob, a block must have been successfully written to the server in an earlier Put Block operation.
+        // refer to https://learn.microsoft.com/en-us/rest/api/storageservices/put-block?tabs=microsoft-entra-id
+        let p = build_abs_path(&self.root, path);
+        let encoded_block_id: String =
+            byte_serialize(BASE64_STANDARD.encode(block_id.to_string()).as_bytes()).collect();
```

Comment:
Please use our existing percent encoding function. And it's better base64 encode the uuid directly without calling to string to make put block_id short.

Reply:
Just use base64 encoding like [here](https://github.com/apache/opendal/blob/93e5f65bbf30df2fed4bdd95bb0685c73c6418c2/core/src/services/azblob/backend.rs#L24)

Reply:
And without `Uuid.to_string()`, this error will occur when complete blockL:
```
ERROR opendal::services: service=azblob operation=Writer::close path=054e2b64-8484-4c82-863f-47f0535a967d written=10485760B -> data close failed: Unexpected (permanent) at Writer::close => AzblobError { code: "InvalidBlockList", message: "The specified block list is invalid. RequestId:b6883402-a73f-4f46-8666-39239d9bae30 Time:2024-02-17T12:54:16.729Z" }
```

Reply:
> Just use base64 encoding like [here](https://github.com/apache/opendal/blob/93e5f65bbf30df2fed4bdd95bb0685c73c6418c2/core/src/services/azblob/backend.rs#L24)

Using our `percent_encode_path` instead of `byte_serialize`.

Reply:
>How about using https://docs.rs/uuid/latest/uuid/struct.Uuid.html#method.simple to make our block_id short?

Just try the `.simple` in commit [1269eea](https://github.com/apache/opendal/pull/4181/commits/1269eea00a2f92dd675a30fb4a93be29d489dddf), but got invalid block list error, so I reverted it.